### DISCUSSION TITLE
Failure on QR decomposition on tensor with extent 1

### DISCUSSION THREAD

**Discussion by PabloAndresCQ** [NVIDIA/cuQuantum#96] [Q&A] [answered]
Dear cuQuantum devs,

I've encountered an edge case on `tensor.decompose` when using `method=tensor.QRMethod()`. Here's the code to reproduce it, using cuQuantum 23.6.0 (for some reason, using `conda install -c conda-forge cuquantum-python` is not installing 23.6.1 for me).

```
import cupy as cp
from cuquantum.cutensornet import tensor

A = cp.ones(shape=(1,2,1))
tensor.decompose(
  "abc->axc,xb",
  A,
  method=tensor.QRMethod()
)
```

I get the following error:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/global/homes/p/pam_q/.conda/envs/py-cuquantum-23.06.0-mypich-py3.9/lib/python3.9/site-packages/cuquantum/cutensornet/tensor.py", line 291, in decompose
    cutn.tensor_qr(handle,
  File "cuquantum/cutensornet/cutensornet.pyx", line 2530, in cuquantum.cutensornet.cutensornet.tensor_qr
  File "cuquantum/cutensornet/cutensornet.pyx", line 2560, in cuquantum.cutensornet.cutensornet.tensor_qr
  File "cuquantum/cutensornet/cutensornet.pyx", line 285, in cuquantum.cutensornet.cutensornet.check_status
cuquantum.cutensornet.cutensornet.cuTensorNetError: CUTENSORNET_STATUS_INTERNAL_ERROR
```

If instead I use SVD I can get the result I was looking for:

```
import cupy as cp
from cuquantum.cutensornet import tensor

A = cp.ones(shape=(1,2,1))
tensor.decompose(
  "abc->axc,xb",
  A,
  method=tensor.SVDMethod(partition="V")
)
```

This returns (as expected):

```
(array([[[-1.]]]), None, array([[-1., -1.]]))
```

I'd appreciate if QRMethod was updated to match the behaviour of SVDMethod. For now I'll use a work around.

Cheers!
Pablo

**Comment by yangcal** [comment]
Hello Pablo,

Thanks for reporting this. We will look into it. For this special case, you may also just skip calling `tensor.decompose` and directly set output Q to a cupy.array with 1 and R as the original tensor data subject to permutation.

Will keep you posted.
Thanks,
Yang

**Reply by yangcal** [reply, marked as answer]
Update: The extent 1 issue for QRMethod has been resolved in the latest cuQuantum python 23.10. Thanks for reporting the issue and we encourage you to try out our latest library.

Thanks,
Yang
